src/modules/kvs/kvs.c

```diff
@@ -2224,10 +2224,18 @@ static int namespace_create (struct kvs_ctx *ctx, const char *ns,
     int len;
     int rv = -1;
 
-    /* If namespace already exists, return EEXIST.  Doesn't matter if
-     * namespace is in process of being removed */
-    if (kvsroot_mgr_lookup_root (ctx->krm, ns)) {
-        errno = EEXIST;
+    /* If namespace already exists, return EEXIST.  If it exists but
+     * is in process of being removed, return EAGAIN instead.
+     */
+    if ((root = kvsroot_mgr_lookup_root (ctx->krm, ns))) {
+        if (root->remove) {
+            (*errmsg) =
+                "namespace with identical name in process "
+                "of being removed. Try again later";
+            errno = EAGAIN;
```

Comment:
I think I would probably return EEXIST in both cases.

My thought process on coming across an EAGAIN might be to wonder whether I'm supposed to retry the call, and if so, after how long, and how many retries?  Then I would look for examples in the code base and wouln't find any.  If we were going to go that way, then I would suggest that we fix the service and not put this on API users.

But I think for now just adding the explanatory message is good enough.